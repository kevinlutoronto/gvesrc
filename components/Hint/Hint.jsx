// @flow
import React from 'react';
import cx from 'classnames';
import { getDataAttrs } from '../../dataUtils';
import type { Data, GlobalAttributes } from '../../types';

import styles from './style.scss';

type Props = {|
  children?: React$Node,
  data?: Data,
  disabled?: boolean,
  error?: boolean,
  text?: string,
  ...GlobalAttributes,
|};

const Hint = ({ children, className, data, disabled, error, text, ...rest }: Props) => {
  const classes = cx(styles.hint, className, {
    [styles.hintDisabled]: disabled,
    [styles.hintError]: error,
  });

  return (
    <div className={classes} {...getDataAttrs(data)} {...rest}>
      {children || text}
    </div>
  );
};

Hint.displayName = 'Plasma@Hint';

export default Hint;
