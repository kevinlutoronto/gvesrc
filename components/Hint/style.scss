@import '../../styles/base';

.hint {
  @extend %type--tiny;

  color: $cGray200;

  &.hintDisabled {
    color: $cGray300;
  }

  &.hintError {
    font-weight: bold;
    color: $cWarningRed;
  }

  > a[href]:only-child {
    font-weight: bold;
    color: $cLinkBlue;
  }
}

* + .hint {
  margin-top: 5px;
}
