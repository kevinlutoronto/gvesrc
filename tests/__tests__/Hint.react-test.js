/* eslint-disable */

import React from 'react';
import Hint from '../../components/Hint/Hint';
import renderer from 'react-test-renderer';

describe('Icon Component', () => {
  test('render', () => {
    const component = renderer.create(<Hint className="test">Hint</Hint>);
    expect(component).toMatchSnapshot();
  });

  test('render with text', () => {
    const component = renderer.create(<Hint text="Hint" />);
    expect(component).toMatchSnapshot();
  });

  test('render disabled', () => {
    const component = renderer.create(<Hint disabled>Hint</Hint>);
    expect(component).toMatchSnapshot();
  });

  test('render error', () => {
    const component = renderer.create(<Hint error>Hint</Hint>);
    expect(component).toMatchSnapshot();
  });
});
