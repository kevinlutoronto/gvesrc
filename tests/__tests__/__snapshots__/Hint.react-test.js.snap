// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon Component render 1`] = `
<div
  className="hint test"
>
  Hint
</div>
`;

exports[`Icon Component render disabled 1`] = `
<div
  className="hint hintDisabled"
>
  Hint
</div>
`;

exports[`Icon Component render error 1`] = `
<div
  className="hint hintError"
>
  Hint
</div>
`;

exports[`Icon Component render with text 1`] = `
<div
  className="hint"
>
  Hint
</div>
`;
